File: kafka-proxy/src/main/java/net/opentsdb/proxy/kafka/KafkaProducer.java
package net.opentsdb.proxy.kafka;

import com.google.inject.Inject;
import com.google.inject.name.Named;
import kafka.javaapi.producer.Producer;
import kafka.javaapi.producer.ProducerData;
import kafka.producer.ProducerConfig;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

import java.util.Properties;

public class KafkaProducer {
  private static final String KAFKA_TOPIC_KEY = "tsdb.kafka.topic";
  private static final Logger logger = LoggerFactory.getLogger(KafkaProducer.class);
  private final Producer<String, String> producer;
  private final String topicId;

  @Inject
  public KafkaProducer(Properties props, @Named(KAFKA_TOPIC_KEY) String topicId) {
    logger.info("Kafka topic: {}", topicId);

    this.topicId = topicId;
    this.producer = new Producer<>(new ProducerConfig(props));
  }

  public void sendMessage(String[] message) {
    StringBuilder result = new StringBuilder();
    for (String aMessage : message) {
      result.append(aMessage).append(" ");
    }

    ProducerData<String, String> data = new ProducerData<>(topicId, result.toString());

    logger.debug("Sending: {}", result.toString());
    producer.send(data);
  }
}

File: opentsdb-proxy/src/main/scala/net/opentsdb/proxy/clients/KafkaProducer.scala
package net.opentsdb.proxy.clients

import com.google.inject.Inject
import java.util.Properties
import org.slf4j.LoggerFactory
import kafka.producer.{KeyedMessage, ProducerConfig, Producer}
import com.google.inject.name.Named

class KafkaProducer @Inject() (props: Properties, @Named("tsdb.kafka.topic") topicId: String) extends Client{
  private final val logger = LoggerFactory.getLogger(classOf[KafkaProducer])
  private final val producer: Producer[String, String] = new Producer(new ProducerConfig(props))

  def sendMessage(message: String) {
    logger.debug("Sending: {}", message)

    producer.send(new KeyedMessage[String, String](topicId, message))
  }
}
